nodes/modifier_change/opensubdivide.py

```diff
@@ -0,0 +1,104 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+
+import bpy
+from bpy.props import IntProperty
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.data_structure import updateNode, match_long_repeat
+
+# from sverchok.utils.modules.ctypes_pyOpenSubdiv import pyOpenSubdiv
+# openSubdivide = pyOpenSubdiv
+
+enable_module = False
+try:
+    import pyOpenSubdiv
+    from pyOpenSubdiv.pysubdivision import pysubdivide    
+    enable_module = True
+except ModuleNotFoundError:
+    enable_module = False
+
+
+from itertools import chain 
+import traceback 
+class SvOpenSubdivideNode(bpy.types.Node,SverchCustomTreeNode):
+    bl_idname = "SvOpenSubdivideNode"
+    bl_label = "OpenSubdiv"
+    bl_icon = 'OUTLINER_OB_EMPTY'
+    sv_icon = None 
+
+    maxlevel : IntProperty(name='level',default=0,min=0,max=5,update=updateNode)
+
+    def sv_init(self,context):
+        self.inputs.new('SvStringsSocket', "Levels").prop_name='maxlevel'
+        self.inputs.new('SvVerticesSocket', "Vertices")
+        self.inputs.new('SvStringsSocket', "Faces")
+
+        self.outputs.new('SvVerticesSocket', "Vertices")
+        self.outputs.new('SvStringsSocket', "Edges")
+        self.outputs.new('SvStringsSocket', "Faces")
```

Comment:
There is sort of convention that data sockets go first and node parameters after them. So input Levels socket should be initialized last. Also, add [sv_internal_links](http://nortikin.github.io/sverchok/docs/contributing/node_api.html#muting) parameter to the node so it could be muted properly. 

I would advise  not to use `prop_name` attribute. This is old inefficient way to add properties to a socket. You can use [socket.use_prop](http://nortikin.github.io/sverchok/docs/contributing/node_api.html#node-sockets) attribute together with setting `socket.default_property_type` to `int`.

Reply:
i can imagine for a property like `maxlevel`, it is desirable to have a lower and upper bound for the possible values that such a slider would accept.

```python
        socket = self.inputs.new('SvStringsSocket', "Levels")
        socket.use_prop = True
        socket.default_property_type = 'int'
        socket.default_int_property = 1
        socket.int_value_range = (1, 6)    # would use get/set behind the scenes
```